sympy/ntheory/primetest.py

```diff
@@ -795,3 +795,39 @@ def is_gaussian_prime(num):
         a = abs(a)
         return isprime(a) and a % 4 == 3
     return isprime(a**2 + b**2)
+
+
+def is_fermat_prime(num):
+    r"""
+    Test if num is a Fermat prime number (True) or not (False).
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory import is_fermat_prime
+    >>> is_fermat_prime(3)
+    True
+    >>> is_fermat_prime(15)
+    False
+    References
+    ==========
+    .. [1] https://oeis.org/A019434
+    .. [2] https://mathworld.wolfram.com/FermatPrime.html
+    .. [3] https://mathworld.wolfram.com/FermatNumber.html
+    """
+
+    num = as_int(num)
+    FERMAT_PRIMES = (3, 5, 17, 257, 65537)
+    
+    if num in FERMAT_PRIMES:
+        return True
+
+    e = num - 1
+    for do in range(2):
+        be = perfect_power(e)
+        if not be:
+            return False
+        b, e = be
+        if b != 2:
+            return False
+    return False if (e < 65537) else None  # it seems unlikely that any more will be found, but not proven
```

Comment:
This is the wrong limit. An actual limit for n in `2**(2**n)-1` that have been tested should be included.

Reply:
[Here](https://www.britannica.com/science/Fermat-prime) they say "it is known that numbers of this form are not prime for values of n from 5 through 30"